src/App.tsx

```diff
@@ -1,27 +1,79 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+enum Sort {
+  ALL = 'all',
+  FIVEFIRST = 'fivefirst',
+  REDGOODS = 'redgoods',
+}
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [modGood, setModGood] = useState<Sort | undefined>(undefined);
```

Comment:
use more meaningful name. Also seems you don't need it.
main point store goods depends from sortType. Sort type changed? Load new goods in `goods` state